ddl/backfilling_dispatcher.go

```diff
@@ -114,14 +106,41 @@ func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context,
 	}
 }
 
-func (*backfillingDispatcherExt) GetNextStep(task *proto.Task) int64 {
+func (h *backfillingDispatcherExt) GetNextStep(
+	taskHandle dispatcher.TaskHandle,
+	task *proto.Task,
+) int64 {
 	switch task.Step {
 	case proto.StepInit:
 		return proto.StepOne
 	case proto.StepOne:
+		// TODO(lance6716): temporary force to use merge step
 		return proto.StepTwo
+
+		// if data files overlaps too much, we need a merge step.
+		subTaskMetas, err := taskHandle.GetPreviousSubtaskMetas(task.ID, proto.StepInit)
+		if err != nil {
+			// TODO(lance6716): should we return error?
+			return proto.StepTwo
```

Comment:
Return the error and retry if it's retriable.

Reply:
I prefer to do it in next pr, this may introduce some conflicts.

Reply:
I think so, we can merge this PR ASAP